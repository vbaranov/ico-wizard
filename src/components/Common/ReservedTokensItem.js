import React from 'react';
import '../../assets/stylesheets/application.css';
import classNames from 'classnames'

const ReservedTokensItem = props => {
  return (
    <div className="reserved-tokens-item-container">
      <div className={classNames('reserved-tokens-item-container-inner', { 'monospace': props.readOnly })}>
        <span className="reserved-tokens-item reserved-tokens-item-left">{props.addr}</span>
        <span className="reserved-tokens-item reserved-tokens-item-middle">{props.dim}</span>
        <span className="reserved-tokens-item reserved-tokens-item-right">{props.val}</span>
      </div>
      {props.readOnly
        ? null
        : <div className="reserved-tokens-item-empty">
          <a onClick={() => props.onRemove(props.num)}>
            <span className="item-remove"/>
          </a>
        </div>
      }
    </div>
  );
};

export default ReservedTokensItem;
