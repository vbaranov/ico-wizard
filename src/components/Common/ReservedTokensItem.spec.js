import React from 'react'
import { StaticRouter } from 'react-router'
import ReservedTokensItem from './ReservedTokensItem'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow } from 'enzyme'

configure({ adapter: new Adapter() })

const token = {
  addr: '0x1111111111111111111111111111111111111111',
  dim: 'percentage',
  val: '12.1234'
}

describe('ReservedTokensItem', () => {
  it(`should render the component with delete button enabled`, () => {
    const wrapper = mount(shallow(
      <ReservedTokensItem
        key={0}
        num={0}
        addr={token.addr}
        dim={token.dim}
        val={token.val}
        readOnly={false}
      />
    ).get(0))

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.reserved-tokens-item-empty')).toHaveLength(1)
  })

  it(`should render the component with no delete button`, () => {
    const wrapper = mount(shallow(
      <ReservedTokensItem
        key={0}
        num={0}
        addr={token.addr}
        dim={token.dim}
        val={token.val}
        readOnly={true}
      />
    ).get(0))

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('reserved-tokens-item-empty')).toHaveLength(0)
  })

  it(`should receive the item index on remove button click`, () => {
    const onRemove = jest.fn()
    const wrapper = mount(
      <ReservedTokensItem
        key={0}
        num={0}
        addr={token.addr}
        dim={token.dim}
        val={token.val}
        readOnly={false}
        onRemove={onRemove}
      />
    )

    wrapper.find('a').simulate('click')

    expect(onRemove).toHaveBeenCalledTimes(1)
    expect(onRemove).toHaveBeenLastCalledWith(0)
  })
})
