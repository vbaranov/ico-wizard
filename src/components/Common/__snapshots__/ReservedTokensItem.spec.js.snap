// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservedTokensItem should render the component with delete button enabled 1`] = `
<div
  className="reserved-tokens-item-container"
>
  <div
    className="reserved-tokens-item-container-inner"
  >
    <span
      className="reserved-tokens-item reserved-tokens-item-left"
    >
      0x1111111111111111111111111111111111111111
    </span>
    <span
      className="reserved-tokens-item reserved-tokens-item-middle"
    >
      percentage
    </span>
    <span
      className="reserved-tokens-item reserved-tokens-item-right"
    >
      12.1234
    </span>
  </div>
  <div
    className="reserved-tokens-item-empty"
  >
    <a
      onClick={[Function]}
    >
      <span
        className="item-remove"
      />
    </a>
  </div>
</div>
`;

exports[`ReservedTokensItem should render the component with no delete button 1`] = `
<div
  className="reserved-tokens-item-container"
>
  <div
    className="reserved-tokens-item-container-inner monospace"
  >
    <span
      className="reserved-tokens-item reserved-tokens-item-left"
    >
      0x1111111111111111111111111111111111111111
    </span>
    <span
      className="reserved-tokens-item reserved-tokens-item-middle"
    >
      percentage
    </span>
    <span
      className="reserved-tokens-item reserved-tokens-item-right"
    >
      12.1234
    </span>
  </div>
</div>
`;
